wfe2/wfe.go

```diff
@@ -2260,16 +2272,30 @@ func (wfe *WebFrontEndImpl) RenewalInfo(ctx context.Context, logEvent *web.Reque
 		return
 	}
 
-	uid := strings.SplitN(request.URL.Path, "/", 3)
-	if len(uid) != 3 {
-		wfe.sendError(response, logEvent, probs.Malformed("Path did not include exactly issuerKeyHash, issuerNameHash, and serialNumber"), nil)
+	// The path prefix has already been stripped, so request.URL.Path here is just
+	// the base64url-encoded DER CertID sequence.
+	der, err := base64.RawURLEncoding.DecodeString(request.URL.Path)
+	if err != nil {
+		wfe.sendError(response, logEvent, probs.Malformed("Path was not base64url-encoded"), nil)
+		return
+	}
+
+	var id certID
+	rest, err := asn1.Unmarshal(der, &id)
+	if err != nil || len(rest) != 0 {
+		wfe.sendError(response, logEvent, probs.Malformed("Path was not a DER-encoded CertID sequence"), nil)
```

Comment:
Similarly, we can return the error from Unmarshal. It's [not documented to return a specific error](https://pkg.go.dev/encoding/asn1#Unmarshal), but I think anything it returns is fair game to show the user. And I suspect a lot of people, for better or for worse, are going to hand-write an encoder for certID, so I expect people will be debugging against this error message a bunch.